poncho/src/poncho/package_serverize.py

```diff
@@ -78,18 +78,13 @@ def main():
                     # see if the user specified an execution method
                     exec_method = event.get("remote_task_exec_method", None)
                     print('Network function: recieved event: {}'.format(event), file=sys.stderr)
-                    if exec_method == "thread":
-                        # create a forked process for function handler
-                        q = queue.Queue()
-                        p = threading.Thread(target=globals()[function_name], args=(event_str, q))
-                        p.start()
-                        p.join()
-                        response = json.dumps(q.get())
-                    elif exec_method == "direct":
+                    if exec_method == "direct":
+                        os.chdir(sandbox)
```

Comment:
Here you want to re-establish the previous working directory after the function is execute, regardless of any error. Please look at  the  `try: except: finally:` statements.

For the fork one this is not needed, as the process that switch working directory exits just after execution.